controllers/storagecluster/external_resources.go

```diff
@@ -369,37 +386,51 @@ func (r *StorageClusterReconciler) createExternalStorageClusterResources(instanc
 }
 
 // createExternalStorageClusterConfigMap creates configmap for external cluster
-func (r *StorageClusterReconciler) createExternalStorageClusterConfigMap(cm *corev1.ConfigMap, found *corev1.ConfigMap, objectKey types.NamespacedName) error {
+func (r *StorageClusterReconciler) createExternalStorageClusterConfigMap(cm, found *corev1.ConfigMap, objectKey types.NamespacedName) error {
 	err := r.Client.Get(context.TODO(), objectKey, found)
 	if err != nil {
 		if errors.IsNotFound(err) {
 			r.Log.Info("Creating External StorageCluster ConfigMap.", "ConfigMap", klog.KRef(objectKey.Namespace, cm.Name))
 			err = r.Client.Create(context.TODO(), cm)
 			if err != nil {
 				r.Log.Error(err, "Creation of External StorageCluster ConfigMap failed.", "ConfigMap", klog.KRef(objectKey.Namespace, cm.Name))
-				return err
 			}
 		} else {
 			r.Log.Error(err, "Unable the get the External StorageCluster ConfigMap.", "ConfigMap", klog.KRef(objectKey.Namespace, cm.Name))
+		}
+		return err
+	}
+	// update the found ConfigMap's Data with the latest changes,
+	// if they don't match
+	if !reflect.DeepEqual(found.Data, cm.Data) {
+		found.Data = cm.DeepCopy().Data
+		if err = r.Client.Update(context.TODO(), found); err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
 // createExternalStorageClusterSecret creates secret for external cluster
-func (r *StorageClusterReconciler) createExternalStorageClusterSecret(sec *corev1.Secret, found *corev1.Secret, objectKey types.NamespacedName) error {
+func (r *StorageClusterReconciler) createExternalStorageClusterSecret(sec, found *corev1.Secret, objectKey types.NamespacedName) error {
```

Comment:
I prefer the original signature - it is easier to read and understand.

Reply:
Done